### ISSUE TITLE
v1.1.0 is not backwards compatible with v1.0.3

### ISSUE THREAD

**Issue by miguelrs** [yohang/Finite#107]
There have been at least 4 method signature changes in new version 1.1.0 that break code that was valid for v1.0.3 - which should only happen in a major version change...

Personally I had problems with:
`StateMachineInterface::apply()`
`PropertiesAwareInterface::get()`
`TransitionInterface::process()`
`CaseStateMachineLoader::supports()`

All those methods have changed their signatures so that a fatal PHP error "Declaration of ... must be compatible with..." will be thrown...

**Comment by RonRademaker** [comment]
This has been discussed in a few PRs, and @yohang considers these breaks acceptable. I've also experienced this and for now I pinned my usage on 1.0.* instead of ~1.0 or ^1.0.

**Comment by tortuetorche** [comment]
So maybe these breaking changes can be mentioned in the `README` or in a `CHANGELOG.md` file?

**Comment by miguelrs** [comment]
You can update anyway and make some little changes in your code to adapt to the new signatures, it's not a big issue... but, those kind of breaks should only happen in a major release - i.e. v.2.0.0

**Comment by RonRademaker** [comment]
Yeah, they're little changes, I just haven't had the time to get to it. A few notes about what to do would be nice though.

**Comment by yohang** [comment]
I'll list them in a `CHANGELOG`.
